vpr/src/route/connection_router.cpp

```diff
@@ -507,21 +593,27 @@ void ConnectionRouter<Heap>::timing_driven_add_to_heap(const t_conn_cost_params
                                       from_node,
                                       to_node,
                                       from_edge,
-                                      target_node);
+                                      target_node,
+                                      run_rcv);
 
     float best_total_cost = rr_node_route_inf_[to_node].path_cost;
     float best_back_cost = rr_node_route_inf_[to_node].backward_path_cost;
 
     float new_total_cost = next.cost;
     float new_back_cost = next.backward_path_cost;
 
-    if (new_total_cost < best_total_cost && new_back_cost < best_back_cost) {
+    if (new_total_cost < best_total_cost && ((new_back_cost < best_back_cost) || (run_rcv))) {
         //Add node to the heap only if the cost via the current partial path is less than the
         //best known cost, since there is no reason for the router to expand more expensive paths.
         //
         //Pre-heap prune to keep the heap small, by not putting paths which are known to be
         //sub-optimal (at this point in time) into the heap.
-        t_heap* next_ptr = heap_.alloc();
+        // Alloc(bool) allocates RCV structures, otherwise they're set to nullptr
+        t_heap* next_ptr;
+        if (run_rcv)
+            next_ptr = heap_.alloc(true);
```

Comment:
Given that there are only two places that "alloc(/*init_data=*/true)" is here and `push_back_node_with_info`, there is very little justification to add an entire virtual method to the HeapInterface to do this.  In this very function, you unnecessarily double allocate because of this design decision.  I believe this demonstrates that design failure here.